ui/apps/platform/src/Containers/BaseImages/BaseImagesModal.test.tsx

```diff
@@ -0,0 +1,175 @@
+        onClose: vi.fn(),
+        onSuccess: vi.fn(),
+    };
+
+    beforeEach(() => {
+        vi.clearAllMocks();
+        mockUseRestMutation.mockReturnValue({
+            mutate: mockMutate,
+            reset: mockReset,
+            isLoading: false,
+            isSuccess: false,
+            isError: false,
+            error: null,
+        });
+    });
+
+    describe('parseBaseImagePath logic', () => {
+        it('should parse simple image path with single colon', async () => {
+            render(<BaseImagesModal {...defaultProps} />);
+
+            const input = screen.getByPlaceholderText(/example-registry.io/);
+            const saveButton = screen.getByRole('button', { name: 'Save' });
+
+            fireEvent.change(input, { target: { value: 'ubuntu:22.04' } });
+            fireEvent.blur(input);
+            fireEvent.click(saveButton);
+
+            await waitFor(() => {
+                expect(mockMutate).toHaveBeenCalledWith(
+                    {
+                        baseImageRepoPath: 'ubuntu',
+                        baseImageTagPattern: '22.04',
+                    },
+                    expect.any(Object)
+                );
+            });
+        });
+
+        it('should parse image path with multiple colons (registry with port)', async () => {
+            render(<BaseImagesModal {...defaultProps} />);
+
+            const input = screen.getByPlaceholderText(/example-registry.io/);
+            const saveButton = screen.getByRole('button', { name: 'Save' });
+
+            fireEvent.change(input, {
+                target: { value: 'docker.io:5000/library/ubuntu:22.04' },
+            });
+            fireEvent.blur(input);
+            fireEvent.click(saveButton);
+
+            await waitFor(() => {
+                expect(mockMutate).toHaveBeenCalledWith(
+                    {
+                        baseImageRepoPath: 'docker.io:5000/library/ubuntu',
+                        baseImageTagPattern: '22.04',
+                    },
+                    expect.any(Object)
+                );
+            });
+        });
+
+        it('should parse image path with tag pattern', async () => {
+            render(<BaseImagesModal {...defaultProps} />);
+
+            const input = screen.getByPlaceholderText(/example-registry.io/);
+            const saveButton = screen.getByRole('button', { name: 'Save' });
+
+            fireEvent.change(input, { target: { value: 'docker.io/library/ubuntu:1.*' } });
+            fireEvent.blur(input);
+            fireEvent.click(saveButton);
+
+            await waitFor(() => {
+                expect(mockMutate).toHaveBeenCalledWith(
+                    {
+                        baseImageRepoPath: 'docker.io/library/ubuntu',
+                        baseImageTagPattern: '1.*',
+                    },
+                    expect.any(Object)
+                );
+            });
+        });
+    });
+
+    describe('form validation', () => {
```

Comment:
These are great use cases to check in component level tests. I'm curious though why go with jest/RTL versus the Cypress component tests we have when it comes to testing render of individual components? I find that the Cypress variant makes debugging easier as you can see when the test executes and you do not have to worry about the jest virtual DOM implementation, `act` errors, etc.

Reply:
Honestly, forgot about Cypress component tests for a second and defaulted to this approach. Makes sense. We can extract the pure functions (parseBaseImagePath and routePaths) into unit tests, and the BaseImagesModal and ImageComponentVulnerabilitiesTable can have Cypress component tests 👍🏼